mongo/errors_test.go

```diff
@@ -0,0 +1,58 @@
+package mongo
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+func TestErrorContains(t *testing.T) {
```

Comment:
Proposed improvements for both test cases:

1. Assert against the exact expected message rather than substrings. IIUC the messages should be deterministic and shouldn't change frequently, so I think checking the exact message is more robust and readable here.
2. Specify multiple errors in the `WriteErrors` array. This will ensure we're correctly stringifying lists of errors (e.g. correctly inserting commas, braces, etc).

Reply:
That sounds reasonable overall. My only concern is that the updated test won't as directly express the important parts of the error message, instead asserting on the whole string. If the error message does end up changing, we can reevaluate the assertion then.

Will update.